### ISSUE TITLE
Client() control scope of resources

### ISSUE THREAD

**Issue by pseudotensor** [dask/dask#6805]
Currently I cannot see way to manage both GPUs and CPUs using dask "resources".

Suppose I have 2 machines each with 1 GPU and 8 cores.  I can set labels for GPU and CPU counts, but dask_cudf/RAPIDS side requires 1 worker/GPU.  But then for any CPU tasks I am suddenly forced to having 1 worker process for an entire node.  I can set nthreads to all cores divided by number of GPUs.  For numpy/panda/scipy things that release GIL that might be kinda ok, but as docs say this is not optimal if something does not release GIL as threads become useless except for I/O.

If I try to add extra workers *just* for CPU resources and have other workers *just* for GPU resources, this does not work either.  Packages like xgboost automatically consume all workers, as the only way to restrict resources currently is to use .compute() or client.submit, but the *client* itself cannot be restricted it seems.  One hits:  https://github.com/dmlc/xgboost/issues/6344  because of this.  If I could call Client() with a "resources=" argument so any use of that client would be using limited resources like .compute, .submit, then that would work.  But AFAIK, I see no such possibility even though would be more general that current scheme of only allowing for .compute, .submit, etc.

I can use 2 schedulers for each type of resource, but that defeats purpose of scheduling and resource management.

Basically, request is for Client() itself to consume resources requests to limit scope of all dask tasks, not just ones that take explicit resources like .compute, client.submit since often we do not use dask in such a fine-grained way, like for xgboost package we have no such control.

**Comment by pseudotensor** [comment]
I'm curious if one work-around is to use the client.submit with resource control and then inside that call do some dask operations.  It seems a bit unwieldy but might work.

**Comment by sjperkins** [comment]
@pseudotensor This is probably a good use case for task [annotations](https://github.com/dask/dask/issues/6701). We've made some progress towards adding [them](https://github.com/dask/dask/pull/6767)

**Comment by sjperkins** [comment]
See https://docs.dask.org/en/latest/api.html#dask.annotate

**Comment by pseudotensor** [comment]
@sjperkins Thanks.  It's seems (to me naively) that just providing a resources to Client() is much more straight-forward than the annotations.  However .compute and .submit is using resources can be used by Client() to give a restricted contextual access to the cluster.
